packages/teleterm/src/services/tshd/middleware.ts

```diff
@@ -65,18 +65,20 @@ export default function middleware<T extends Record<string, any>>(
   });
 }
 
-export const withLogging = (logger: Logger) => {
-  const interceptor: UnaryInterceptor = (options: any, nextCall: Function) => {
+export const withLogging = (logger: Logger): UnaryInterceptor => {
+  return (options, nextCall) => {
     const method = options.method_definition.path;
-    const params = {
-      start(metadata: any, listener: any, next: Function) {
+    const filterPasswords = object =>
+      filterSensitiveProperties(object, ['passw']);
```

Comment:
How do you feel about defining `filterPasswords` outside of `withLogging` or not parameterizing `filterSensitiveProperties` at all (by defining `filteredProperties` once in the scope of the file) and using `filterSensitiveProperties` instead of `filterPasswords`?

I know that this style is a popular convention inside React components. But in general I think that from the perspective of code maintainability, it's best to define a function in the outermost possible scope.

When I see `filterPasswords` defined within the scope of the function returned from `withLogging`, I assume that it's because `filterPasswords` needs something from that scope, when actually this is not the case.

Maybe it's just that I have wrong heuristics here, but I believe that it's best to clearly communicate the boundaries of functions, what they do need and what they don't need.

Reply:
Ah, good intuition, we don't need that parameter.
I defined `filteredProperties` inside the function (I like that style more than using vars in the file scope), I guess it is fine?

Reply:
Why not define it in the file scope though? 👀 This way you clearly communicate that it won't change during the lifetime of the program. Also, this avoids creating a new array each time the function is called (although idk, maybe modern JS engines are able to optimize that away).

I think it's perfectly okay to define vars in the file scope, it's almost like a constant since there's never a need to change it during the lifetime of the program. You could even uppercase it, from what I've seen we use this convention in our codebase:

```
$ rg "^const [A-Z]{3,}" --pretty --color=never | head
packages/teleterm/src/mainProcess/runtimeSettings.ts
10:const RESOURCES_PATH = app.isPackaged

packages/build/devserver/index.js
39:const PROXY_TARGET = urlObj.host;
40:const ROOT = '/web';
41:const PORT = 8080;

packages/teleterm/src/ui/DocumentTerminal/Terminal/ctrl.ts
25:const WINDOW_RESIZE_DEBOUNCE_DELAY = 200;